runtime/domain.c

```diff
@@ -533,8 +529,7 @@ static void* domain_thread_func(void* v)
     caml_gc_log("Domain starting (unique_id = %"ARCH_INTNAT_PRINTF_FORMAT"u)",
                 domain_self->interruptor.unique_id);
     caml_domain_start_hook();
-    caml_callback(caml_read_root(callback), Val_unit);
-    caml_delete_root(callback);
+    caml_callback(*domain_callback, Val_unit);
     domain_terminate();
```

Comment:
I think we need to do something to release the resources associated with `domain_callback` otherwise we will leak here. 

My reasoning is that the function `caml_domain_spawn` has allocated `domain_callback` with `caml_stat_alloc_noexc` and registered it with the runtime with `caml_register_generational_global_root`, the resource ownership is passed to the child thread when it signals `Dom_started` back to the parent.  

Don't we need to have this just before the call to `domain_terminate`:
```C
    caml_remove_generational_global_root(domain_callback);
    caml_stat_free(domain_callback);
```

Reply:
Ah yes, missed it, thanks for spotting this!